### PR TITLE
Reimplement pandas deconstruction

### LINKED ISSUE #1158: Perspective-Python does not support DataFrames with categorical axis
### Steps to Reproduce:
 1. Create a dataframe with a categorical axis.
 2. Attempt to load into a `PerspectiveWidget` or `Table`.

### Expected Result:
A valid widget or Table.

### Actual Result:

```python
---------------------------------------------------------------------------
PerspectiveError                          Traceback (most recent call last)
<ipython-input-2-acefd98973df> in <module>
      3     "b": [1, 2, 3, 4, 5]
      4 })
----> 5 perspective.PerspectiveWidget(df)

~/work/perspective/python/perspective/perspective/widget/widget.py in __init__(self, table_or_data, index, limit, client, **kwargs)
    263                     load_kwargs.update({"limit": limit})
    264 
--> 265                 self.load(table_or_data, **load_kwargs)
    266 
    267     def load(self, data, **options):

~/work/perspective/python/perspective/perspective/widget/widget.py in load(self, data, **options)
    277             self._data = d
    278         else:
--> 279             super(PerspectiveWidget, self).load(data, **options)
    280 
    281         # proactively notify front-end of new data

~/work/perspective/python/perspective/perspective/viewer/viewer.py in load(self, table_or_data, **options)
    159             table = table_or_data
    160         else:
--> 161             table = Table(table_or_data, **options)
    162 
    163         self.manager.host_table(name, table)

~/work/perspective/python/perspective/perspective/table/table.py in __init__(self, data, limit, index)
     47             _accessor = data
     48         else:
---> 49             _accessor = _PerspectiveAccessor(data)
     50 
     51         self._date_validator = _PerspectiveDateValidator()

~/work/perspective/python/perspective/perspective/table/_accessor.py in __init__(self, data_or_schema)
    111 
    112                 if not isinstance(array, numpy.ndarray):
--> 113                     raise PerspectiveError("Mixed datasets of numpy.ndarray and lists are not supported.")
    114 
    115                 dtype = array.dtype

PerspectiveError: Mixed datasets of numpy.ndarray and lists are not supported.
```

Screen Shot 2020-08-19 at 1 17 40 PM

### Environment:

Latest master 

### Additional Context:

Loading works fine with the column converted to string:

Screen Shot 2020-08-19 at 1 22 22 PM

and converted to dict:

Screen Shot 2020-08-19 at 1 22 40 PM

Looks like a problem in `_accessor.py` and our Pandas loading code.

### LINKED ISSUE #1243: Filter options are not converted to json-serializeable values
### Steps to Reproduce:
p = PerspectiveWidget(...)
p.filters = [['Col', '==', datetime(2020, 10, 30)]]

### Expected Result:
The filter to be applied.

### Actual Result:
ipywidgets error about datetime not being json-serializeable 

### Environment:
All python environments

### Additional Context:
fixed in #1238

### PR DESCRIPTION
**Pull request by timkpaine**
Pivoted Pandas dataframe deconstruction is fundamentally broken in our current state, as exhibited by a few issues (#1122, #1123, etc). This PR rectifies that problem by better deconstructing pivoted data frames.

Also changed:
- Certain pivoted data frames had ' ' columns for visual style, this is changed to 'value' to be more sensible.
- Date/Datetime arguments to filter are converted to their string representation (unrelated to pandas, fixes #1243)
- upgrade pip for release image issue (potential fix for #1244)
- remove commented Mojave code (broken azure pipelines environment)
- fixes #1158

Notebooks and test cases for new pandas behavior updated/added.

### CODE REVIEW

**Review by sc1f — CHANGES_REQUESTED** [review]
Can we split the commits where we run `black` into a separate PR? There's too much reformatting noise to discern the real changes as part of this PR.

File: python/perspective/perspective/tests/widget/test_widget_pandas.py [hunk 1]
```diff
@@ -16,108 +16,427 @@
...
+        )
         widget = PerspectiveWidget(df_both, columns=["first", "third"])
-        assert widget.columns == ['first', "third"]
-        assert widget.column_pivots == ['first', 'second', 'third']
-        assert widget.row_pivots == ['index']
+        assert widget.columns == ["first", "third"]
+        assert widget.column_pivots == ["first", "second", "third"]
+        assert widget.row_pivots == ["index"]
+
+    def test_pivottable_values_index(self):
```
Comment (timkpaine) [inline python/perspective/perspective/tests/widget/test_widget_pandas.py:363]: these are the only changes, addition of 2 tests

**Review by sc1f — APPROVED** [review]
Checked out locally - changes, examples and reformatting look good. @texodus we should probably merge this first and rebase, or manually split it into two PRs just so that other PRs aren't mired in merge issues.

Thanks for fixing the datetime issue! It is only a fix for the Python widget side - for a more comprehensive fix we need to target the date parser in JS/C++ and make it more user-friendly.

**Comment by texodus** [comment]
Looks good!  Thanks for the PR!

**Pull request merged**